### ISSUE TITLE
[question] Qt Private components

### ISSUE THREAD

**Issue by Jihadist** [conan-io/conan-center-index#5841]
How can i use Qt private classes? 
With official Qt binaries i do it this way
```
find_package(QT NAMES Qt6 Qt5 COMPONENTS Core Gui REQUIRED)
find_package(Qt${QT_VERSION_MAJOR} COMPONENTS Core Gui REQUIRED)
...
target_link_libraries(${PROJECT_NAME}
 Qt${QT_VERSION_MAJOR}::Core
 Qt${QT_VERSION_MAJOR}::GuiPrivate
 )
```
But i have errors with conan binaries. So how can i legal import conan qt private headers into my project?
```error: C1083: Cannot open include file: 'private/q*_p.h': No such file or directory```

For example this works with offical installation:
```#include <private/qzipreader_p.h>``` but not with conan
i can include with conan qt it this way ```#include "5.15.2/QtGui/private/qzipreader_p.h"``` so conan qt provides this headers

**Comment by ericLemanissier** [comment]
As you realized, private targets are not create by the recipe currently. We could consider adding them, but this requires some thought because you loose all the binary compatibility guaranties when using private parts of qt. In the end this goes against what conan (or any package manager) wants to provide, so this is not something we want to be easy to do.

**Comment by Jihadist** [comment]
>  this requires some thought because you loose all the binary compatibility guaranties when using private parts of qt

I know this but should i do with libs depend on qt private? For example https://github.com/QtExcel/QXlsx or another example currently exists in conan - https://github.com/conan-io/conan-center-index/tree/master/recipes/qtxlsxwriter with depends on private component https://github.com/dbzhang800/QtXlsxWriter/search?q=QZipReader

**Comment by ericLemanissier** [comment]
that's a valid point. we will certainly have to create these private components

**Comment by ericLemanissier** [comment]
@Jihadist can you please try with https://github.com/conan-io/conan-center-index/pull/6301 ? it should create the GuiPrivate targets (qt6 only)

**Comment by Jihadist** [comment]
@ericLemanissier will QtCorePrivate work without QtGui module? Because Qt provides GuiPrivate and CorePrivate component.

**Comment by ericLemanissier** [comment]
yes, QtCorePrivate was added in https://github.com/conan-io/conan-center-index/pull/5218

**Comment by Jihadist** [comment]
> 
> 
> yes, QtCorePrivate was added in #5218

Is there any chances for qt5 support? Because too many projects use qt5 to ignore it.

**Comment by ericLemanissier** [comment]
of course, but I'd like to have some feedback on the existing implementation before porting it to qt5

**Comment by ohanar** [comment]
Sadly this is another blocking issue that is preventing us from migrating off of the bincrafters qt5 recipe.

**Comment by ericLemanissier** [comment]
@ohanar can you please try with https://github.com/conan-io/conan-center-index/pull/6301 ? it has GuiPrivate module

**Comment by ohanar** [comment]
@ericLemanissier No, that is only qt6.

**Comment by ericLemanissier** [comment]
I just ported the private components to qt5 in https://github.com/conan-io/conan-center-index/pull/6376. feedback is welcome !

**Comment by ohanar** [comment]
@ericLemanissier Sorry, got busy with other things and only recently was able to give it a try. Yes, #6376 does resolve this issue for me.

**Comment by ruilvo** [comment]
Fixed by #6376
(To trigger the cleanup)
